### ISSUE TITLE
`pl.read_excel` only reads visible number of decimals

### ISSUE THREAD

**Issue by mcrumiller** [pola-rs/polars#12052]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

Setup: Excel book with the exact same number repeated three times, but the visible format is set to 1, 2, and 5 decimals:

#### `test.xlsx`
![image](https://github.com/pola-rs/polars/assets/1896992/e5cd7921-99b6-469f-8be3-9126f69bc172)

```shell
>>> import polars as pl
>>> pl.read_excel("test.xlsx")
shape: (3, 2)
┌─────┬─────────┐
│ A   ┆ B       │
│ --- ┆ ---     │
│ i64 ┆ f64     │
╞═════╪═════════╡
│ 1   ┆ 1.2     │
│ 2   ┆ 1.23    │
│ 3   ┆ 1.23457 │
└─────┴─────────┘
```

### Issue description

`pl.read_excel` ignores non-visible decimals. This is pretty unintuitive, and while it's generally a problem with `xlsx2csv` and can be fixed by providing `xlsx2csv_options={"floatformat": "%f"}`, it's a bit buried in the documentation--in fact, it's not any where in the polars documentation, and its resolution requires finding the `xlsx2csv` documentation and discovering that the `floatformat` property exists.

I think we should supply `%f` as the default in `read_excel` when using the `xlsx2csv` engine.

### Expected behavior

```shell
>>> pl.read_excel("test.xlsx", xlsx2csv_options={"floatformat": "%f"})
shape: (3, 2)     
┌─────┬──────────┐
│ A   ┆ B        │
│ --- ┆ ---      │
│ i64 ┆ f64      │
╞═════╪══════════╡
│ 1   ┆ 1.234567 │
│ 2   ┆ 1.234567 │
│ 3   ┆ 1.234567 │
└─────┴──────────┘
```

### Installed versions

<details>

```
--------Version info---------
Polars:              0.19.11
Index type:          UInt32
Platform:            Windows-10-10.0.19045-SP0
Python:              3.11.2 (tags/v3.11.2:878ead1, Feb  7 2023, 16:38:35) [MSC v.1934 64 bit (AMD64)]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
cloudpickle:         <not installed>
connectorx:          0.3.2
deltalake:           <not installed>
fsspec:              <not installed>
gevent:              <not installed>
matplotlib:          3.7.1
numpy:               1.26.1
openpyxl:            3.1.2
pandas:              2.1.1
pyarrow:             11.0.0
pydantic:            <not installed>
pyiceberg:           <not installed>
pyxlsb:              <not installed>
sqlalchemy:          2.0.7
xlsx2csv:            0.8.1
xlsxwriter:          3.0.9
```

</details>

**Comment by mcrumiller** [comment]
@alexander-beedie

**Comment by alexander-beedie** [comment]
Closed by #12081.
